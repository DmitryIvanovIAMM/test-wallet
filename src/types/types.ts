export interface TransactionType {
  Payment: 'Payment';
  Credit: 'Credit';
}

export interface Transaction {
  _id: string;
  transactionType: TransactionType;
  amount: number;
  currency: string;
  transactionName: string;
  transactionDescription: string;
  date: Date | string;
}