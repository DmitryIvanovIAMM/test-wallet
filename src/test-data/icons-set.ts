import { library } from '@fortawesome/fontawesome-svg-core';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import {
  faApple,
  faGoogle,
  faFacebook,
  faGithub,
  faDiscord,
  faYoutube,
  faFigma,
  faMicrosoft,
  faSquareReddit,
  faMendeley,
  faCcVisa,
  faCcAmazonPay
} from '@fortawesome/free-brands-svg-icons';

library.add([
  faApple,
  faGoogle,
  faFacebook,
  faGithub,
  faDiscord,
  faYoutube,
  faFigma,
  faMicrosoft,
  faSquareReddit,
  faMendeley,
  faCcAmazonPay
]);

export const IconsMapper: { [name: string]: IconDefinition } = {
  Apple: faApple,
  Google: faGoogle,
  Facebook: faFacebook,
  Github: faGithub,
  Discord: faDiscord,
  Youtube: faYoutube,
  Figma: faFigma,
  Microsoft: faMicrosoft,
  Reddit: faSquareReddit,
  Mendeley: faMendeley,
  Amazon: faCcAmazonPay,
  Visa: faCcVisa
};

export const getIconDefinitionByBrand = (iconName: string): IconDefinition => {
  return IconsMapper[iconName] || IconsMapper['Visa'];
};
