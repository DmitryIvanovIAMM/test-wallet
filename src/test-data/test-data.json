{
  "test-data": [
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3b1",
      "transactionType": "Payment",
      "amount": 100,
      "currency": "USD",
      "transactionName": "Facebook",
      "transactionDescription": "Payment for the service",
      "date": "2020-09-14T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3b2",
      "transactionType": "Credit",
      "amount": 200,
      "currency": "USD",
      "transactionName": "Amazon",
      "transactionDescription": "Credit for the product",
      "date": "2020-09-15T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3b3",
      "transactionType": "Payment",
      "amount": 150,
      "currency": "USD",
      "transactionName": "Discord",
      "transactionDescription": "Payment for the service",
      "date": "2020-09-16T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3b4",
      "transactionType": "Credit",
      "amount": 250,
      "currency": "USD",
      "transactionName": "Yotube",
      "transactionDescription": "Credit for the product",
      "date": "2020-09-17T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3b5",
      "transactionType": "Payment",
      "amount": 300,
      "currency": "USD",
      "transactionName": "Yotube",
      "transactionDescription": "Payment for the service",
      "date": "2020-09-18T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3b6",
      "transactionType": "Credit",
      "amount": 350,
      "currency": "USD",
      "transactionName": "Costco",
      "transactionDescription": "Credit for the product",
      "date": "2020-09-19T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3b7",
      "transactionType": "Payment",
      "amount": 400,
      "currency": "USD",
      "transactionName": "Yotube",
      "transactionDescription": "Payment for the service",
      "date": "2020-09-20T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3b8",
      "transactionType": "Credit",
      "amount": 450,
      "currency": "USD",
      "transactionName": "Figma",
      "transactionDescription": "Credit for the product",
      "date": "2020-09-21T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3b9",
      "transactionType": "Payment",
      "amount": 500,
      "currency": "USD",
      "transactionName": "Apple",
      "transactionDescription": "Payment for the service",
      "date": "2020-09-22T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3ba",
      "transactionType": "Credit",
      "amount": 550,
      "currency": "USD",
      "transactionName": "Microsoft",
      "transactionDescription": "Credit for the product",
      "date": "2020-09-23T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3bb",
      "transactionType": "Payment",
      "amount": 600,
      "currency": "USD",
      "transactionName": "Google",
      "transactionDescription": "Payment for the service",
      "date": "2020-09-24T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3bc",
      "transactionType": "Credit",
      "amount": 650,
      "currency": "USD",
      "transactionName": "Facebook",
      "transactionDescription": "Credit for the product",
      "date": "2020-09-25T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3bd",
      "transactionType": "Payment",
      "amount": 700,
      "currency": "USD",
      "transactionName": "Twitter",
      "transactionDescription": "Payment for the service",
      "date": "2020-09-26T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3be",
      "transactionType": "Credit",
      "amount": 750,
      "currency": "USD",
      "transactionName": "LinkedIn",
      "transactionDescription": "Credit for the product",
      "date": "2020-09-27T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3bf",
      "transactionType": "Payment",
      "amount": 800,
      "currency": "USD",
      "transactionName": "Netflix",
      "transactionDescription": "Payment for the service",
      "date": "2020-09-28T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c0",
      "transactionType": "Credit",
      "amount": 850,
      "currency": "USD",
      "transactionName": "Spotify",
      "transactionDescription": "Credit for the product",
      "date": "2020-09-29T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c1",
      "transactionType": "Payment",
      "amount": 900,
      "currency": "USD",
      "transactionName": "Uber",
      "transactionDescription": "Payment for the service",
      "date": "2020-09-30T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c2",
      "transactionType": "Credit",
      "amount": 950,
      "currency": "USD",
      "transactionName": "Lyft",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-01T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c3",
      "transactionType": "Payment",
      "amount": 1000,
      "currency": "USD",
      "transactionName": "Airbnb",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-02T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c4",
      "transactionType": "Credit",
      "amount": 1050,
      "currency": "USD",
      "transactionName": "Dropbox",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-03T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c5",
      "transactionType": "Payment",
      "amount": 1100,
      "currency": "USD",
      "transactionName": "Slack",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-04T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c6",
      "transactionType": "Credit",
      "amount": 1150,
      "currency": "USD",
      "transactionName": "Zoom",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-05T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c7",
      "transactionType": "Payment",
      "amount": 1200,
      "currency": "USD",
      "transactionName": "Adobe",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-06T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c8",
      "transactionType": "Credit",
      "amount": 1250,
      "currency": "USD",
      "transactionName": "Salesforce",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-07T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3c9",
      "transactionType": "Payment",
      "amount": 1300,
      "currency": "USD",
      "transactionName": "Oracle",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-08T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3ca",
      "transactionType": "Credit",
      "amount": 1350,
      "currency": "USD",
      "transactionName": "SAP",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-09T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3cb",
      "transactionType": "Payment",
      "amount": 1400,
      "currency": "USD",
      "transactionName": "IBM",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-10T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3cc",
      "transactionType": "Credit",
      "amount": 1450,
      "currency": "USD",
      "transactionName": "Intel",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-11T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3cd",
      "transactionType": "Payment",
      "amount": 1500,
      "currency": "USD",
      "transactionName": "AMD",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-12T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3ce",
      "transactionType": "Credit",
      "amount": 1550,
      "currency": "USD",
      "transactionName": "NVIDIA",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-13T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3cf",
      "transactionType": "Payment",
      "amount": 1600,
      "currency": "USD",
      "transactionName": "Tesla",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-14T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d0",
      "transactionType": "Credit",
      "amount": 1650,
      "currency": "USD",
      "transactionName": "SpaceX",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-15T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d1",
      "transactionType": "Payment",
      "amount": 1700,
      "currency": "USD",
      "transactionName": "Blue Origin",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-16T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d2",
      "transactionType": "Credit",
      "amount": 1750,
      "currency": "USD",
      "transactionName": "Virgin Galactic",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-17T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d3",
      "transactionType": "Payment",
      "amount": 1800,
      "currency": "USD",
      "transactionName": "Boeing",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-18T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d4",
      "transactionType": "Credit",
      "amount": 1850,
      "currency": "USD",
      "transactionName": "Airbus",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-19T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d5",
      "transactionType": "Payment",
      "amount": 1900,
      "currency": "USD",
      "transactionName": "Lockheed Martin",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-20T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d6",
      "transactionType": "Credit",
      "amount": 1950,
      "currency": "USD",
      "transactionName": "Northrop Grumman",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-21T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d7",
      "transactionType": "Payment",
      "amount": 2000,
      "currency": "USD",
      "transactionName": "Raytheon",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-22T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d8",
      "transactionType": "Credit",
      "amount": 2050,
      "currency": "USD",
      "transactionName": "General Dynamics",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-23T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3d9",
      "transactionType": "Payment",
      "amount": 2100,
      "currency": "USD",
      "transactionName": "BAE Systems",
      "transactionDescription": "Payment for the service",
      "date": "2020-10-24T00:00:00.000Z"
    },
    {
      "_id": "5f5f3b3b9b1f3b0017f3b3da",
      "transactionType": "Credit",
      "amount": 2150,
      "currency": "USD",
      "transactionName": "Thales Group",
      "transactionDescription": "Credit for the product",
      "date": "2020-10-25T00:00:00.000Z"
    }
  ]
}
