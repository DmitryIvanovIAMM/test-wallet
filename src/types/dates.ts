export const daysOfWeek = [
  'Sunday',
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday'
];

export const getDateString = (date: Date | string) => {
  const incomingDate = new Date(date);
  const lastWeekDate = new Date(new Date().getTime() - 7 * 24 * 60 * 60 * 1000);

  return incomingDate > lastWeekDate
    ? daysOfWeek[incomingDate.getDay()]
    : incomingDate.toLocaleDateString();
};
