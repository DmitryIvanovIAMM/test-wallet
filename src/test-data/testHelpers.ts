import { Transaction, TransactionType, User } from '@/types/types';

export const authorizesUser: User = {
  _id: '5f5f3b3b9b1f3b0017f3',
  name: 'John Doe',
  email: 'test.email1@email.com'
};
export const testPaymentTransaction: Transaction = {
  _id: '5f5f3b3b9b1f3b0017f3',
  transactionType: TransactionType.Payment,
  amount: 100,
  currency: 'USD',
  transactionName: 'Payment',
  transactionDescription: 'Test transaction 1',
  isPending: false,
  authorizedUser: authorizesUser,
  date: new Date('2020-09-14T00:00:00.000Z')
};
export const testCreditTransaction: Transaction = {
  _id: '5f5f3b3b9b1f3b0017f3',
  transactionType: TransactionType.Credit,
  amount: 100,
  currency: 'USD',
  transactionName: 'Credit',
  transactionDescription: 'Test transaction 2',
  isPending: false,
  authorizedUser: authorizesUser,
  date: new Date('2020-09-15T00:00:00.000Z')
};
