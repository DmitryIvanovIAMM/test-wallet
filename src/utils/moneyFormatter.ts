import { TransactionType } from '@/types/types';

export const moneyFormatterWithTransactionType = (
  amount: number,
  currency: string,
  transactionType: TransactionType
) => {
  const formattedAmount = moneyFormatter(amount, currency);

  if (transactionType === TransactionType.Payment) {
    return `+${formattedAmount}`;
  } else {
    return formattedAmount;
  }
};

export const moneyFormatter = (amount: number, currency: string) => {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: currency,
    minimumFractionDigits: 2
  }).format(amount);
};

export const randomIntFromInterval = (min: number, max: number): number => {
  // min and max included
  return Math.floor(Math.random() * (max - min + 1) + min);
};
