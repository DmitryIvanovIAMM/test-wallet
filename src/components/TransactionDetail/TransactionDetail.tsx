'use client';

import { Transaction } from '@/types/types';
import { moneyFormatter } from '@/utils/moneyFormatter';
import { getDateTimeString } from '@/types/dates';
import Link from 'next/link';

export interface TransactionDetailProps {
  transaction: Transaction;
}
export default function TransactionDetail({ transaction }: TransactionDetailProps) {
  return (
    <div
      className="flex flex-col gap-4 w-full h-screen p-2 xs:m-20 bg-gray-200 m-0 justify-start"
      data-testid="transaction-detail"
    >
      <div className="col-span-1 h-8 align-top">
        <button className="w-full text-blue-500 bg-gray-200 hover:bg-gray-200" onClick={() => {}}>
          <Link
            className="w-full inline-block align-text-bottom text-left"
            data-testid="transaction-detail-button"
            href={`/`}
          >
            {'<'}
          </Link>
        </button>
      </div>
      <div className="font-bold text-4xl w-full text-center mt-5 text-gray-800">
        {moneyFormatter(transaction.amount, transaction.currency)}
      </div>
      <div className="text-sm text-gray-600 align-middle text-center">
        {transaction.transactionName}
      </div>
      <div className="text-sm text-gray-600 align-middle text-center">
        {getDateTimeString(transaction.date)}
      </div>
      <div className="grid grid-cols-1 gap-1 rounded-md w-full p-4  bg-white mt-5">
        <div className="font-bold text-sm justify-center text-gray-800">
          {`Status: ${transaction.isPending ? 'Pending' : 'Approved'}`}
        </div>
        <div className="text-sm text-gray-600">{transaction.transactionDescription}</div>
        <hr />
        <div className="grid grid-cols-2 justify-between font-bold text-sm text-gray-800">
          <div>Total</div>
          <div className="text-right">
            {moneyFormatter(transaction.amount, transaction.currency)}
          </div>
        </div>
      </div>
    </div>
  );
}
