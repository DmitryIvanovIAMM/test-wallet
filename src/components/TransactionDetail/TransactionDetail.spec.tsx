import { render } from '@testing-library/react';
import { moneyFormatter } from '@/utils/moneyFormatter';
import { getDateTimeString } from '@/types/dates';
import { testPaymentTransaction } from '@/test-data/testHelpers';
import TransactionDetail, {
  TransactionDetailProps
} from '@/components/TransactionDetail/TransactionDetail';

const setup = (propsOverride: Partial<TransactionDetailProps> = {}) => {
  const props: TransactionDetailProps = {
    transaction: testPaymentTransaction,
    ...propsOverride
  };

  const container = render(<TransactionDetail {...props} />);

  return {
    props,
    container
  };
};

describe('TransactionDetail component should', () => {
  it('render correctly', async () => {
    const { container } = await setup();
    expect(container).toMatchSnapshot();
  });

  it('should show transaction details', async () => {
    const { container, props } = await setup();

    expect(container.getByText(props.transaction.transactionName)).toBeInTheDocument();
    expect(
      container.queryAllByText(moneyFormatter(props.transaction.amount, props.transaction.currency))
        .length
    ).toBe(2);
    expect(container.getByText(getDateTimeString(props.transaction.date))).toBeInTheDocument();
    expect(container.getByText(props.transaction.transactionDescription)).toBeInTheDocument();
  });
});
