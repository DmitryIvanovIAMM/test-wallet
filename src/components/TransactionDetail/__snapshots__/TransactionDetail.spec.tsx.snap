// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetail component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-4 w-full h-screen p-2 xs:m-20 bg-gray-200 m-0 justify-start"
        data-testid="transaction-detail"
      >
        <div
          class="col-span-1 h-8 align-top"
        >
          <button
            class="w-full text-blue-500 bg-gray-200 hover:bg-gray-200"
          >
            <a
              class="w-full inline-block align-text-bottom text-left"
              data-testid="transaction-detail-button"
              href="/"
            >
              &lt;
            </a>
          </button>
        </div>
        <div
          class="font-bold text-4xl w-full text-center mt-5 text-gray-800"
        >
          $100.00
        </div>
        <div
          class="text-sm text-gray-600 align-middle text-center"
        >
          Payment
        </div>
        <div
          class="text-sm text-gray-600 align-middle text-center"
        >
          9/14/2020, 24:00
        </div>
        <div
          class="grid grid-cols-1 gap-1 rounded-md w-full p-4  bg-white mt-5"
        >
          <div
            class="font-bold text-sm justify-center text-gray-800"
          >
            Status: Approved
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Test transaction 1
          </div>
          <hr />
          <div
            class="grid grid-cols-2 justify-between font-bold text-sm text-gray-800"
          >
            <div>
              Total
            </div>
            <div
              class="text-right"
            >
              $100.00
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-4 w-full h-screen p-2 xs:m-20 bg-gray-200 m-0 justify-start"
      data-testid="transaction-detail"
    >
      <div
        class="col-span-1 h-8 align-top"
      >
        <button
          class="w-full text-blue-500 bg-gray-200 hover:bg-gray-200"
        >
          <a
            class="w-full inline-block align-text-bottom text-left"
            data-testid="transaction-detail-button"
            href="/"
          >
            &lt;
          </a>
        </button>
      </div>
      <div
        class="font-bold text-4xl w-full text-center mt-5 text-gray-800"
      >
        $100.00
      </div>
      <div
        class="text-sm text-gray-600 align-middle text-center"
      >
        Payment
      </div>
      <div
        class="text-sm text-gray-600 align-middle text-center"
      >
        9/14/2020, 24:00
      </div>
      <div
        class="grid grid-cols-1 gap-1 rounded-md w-full p-4  bg-white mt-5"
      >
        <div
          class="font-bold text-sm justify-center text-gray-800"
        >
          Status: Approved
        </div>
        <div
          class="text-sm text-gray-600"
        >
          Test transaction 1
        </div>
        <hr />
        <div
          class="grid grid-cols-2 justify-between font-bold text-sm text-gray-800"
        >
          <div>
            Total
          </div>
          <div
            class="text-right"
          >
            $100.00
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
